design/external-authorization-design.md

```diff
@@ -0,0 +1,715 @@
+# External Authorization Support
+
+Status: Reviewing
+
+## Abstract
+This document describes a design for performing request authorization for virtual hosts hosted by Contour.
+
+## Background
+
+- [#432][5] Provide support for integrating with an external auth service.
+- [#2459][3] External auth background and goals.
+- [#2325][9] Failures in External services configured in envoy should be visible to the application developer.
+
+## Goals
+- Allow operators to integrate existing authorization services with Contour.
+- Allow arbitrary types of authentication and authorization to be supported.
+- Decouple Contour from external authorization, so it can evolve at an independent rate.
+- Integrate cleanly with the HTTPProxy API.
+
+## Non Goals
```

Comment:
Also supporting v1/v1beta1.Ingress is not a goal?

Reply:
Yep, so far it is not a goal. In general, when adding features, we haven't also added them to Ingress, so not sure if we want to do that here 🤷

Reply:
I think that we will add these features to HTTPProxy first, and consider if we want to add them to Ingress later. The only way we could do that would be with a forest of annotations, which I am reluctant to do without a good reason.

Reply:
I'm not suggesting that we add, just to call it out in the design so when someone comes and reads the design doc, it's clear. That was one of the original goals of these designs. It was to plan out how or what a feature looked like, even if parts were far off in the future. 

The second goal was to have these features planned out so that more users could contribute with a plan already in place.